torchrl/modules/tensordict_module/world_models.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from math import sqrt
+
+import torch
+import torch.distributions as d
+import torch.nn as nn
+import torch.nn.functional as F
+
+from torchrl.modules.tensordict_module import TensorDictModule, TensorDictSequence
+
+__all__ = ["WorldModelWrapper"]
+
+
+class WorldModelWrapper(TensorDictSequence):
+    """
+    World model wrapper.
+    This module wraps together a world model and a reward model.
+    The world model is used to predict an imaginary world state.
+    The reward model is used to predict the reward of the imaginary world state.
+
+    Args:
+        world_model (TensorDictModule): a world model that generates a world state.
+        reward_model (TensorDictModule): a reward model, that reads the world state and returns a reward
+
+    """
+
+    def __init__(
+        self,
+        world_modeler_operator: TensorDictModule,
```

Comment:
is modeleR intended?

Reply:
Also we have a wrapper which is a sequence,
but has the name of one of the components (`WorldModelWrapper` matches `world_model`), so the names are a bit confusing

Reply:
yes world modeler is intended to make the distinction with world model which comports the reward modeling. I however think that it could be good to find a better name but i haven't found any atm.